RIOT-OS/RIOT
# boards/same51-curiosity-nano: add support for the board

### Contribution description

The [SAM E51 Curiosity Nano](https://www.microchip.com/en-us/development-tool/ev76s68a) is a cheap evaluation board for the SAM E51 family of MCUs. This family is pretty much identical with the SAM E54 family, except that it does not feature an Ethernet peripheral.

It does however feature two CAN peripherals.


### Testing procedure

<details><summary>tests/periph/uart</summary>

```
2025-01-07 07:04:37,247 # main(): This is RIOT! (Version: 2025.01-devel-342-gfad857-cpu/same51)
2025-01-07 07:04:37,247 # 
2025-01-07 07:04:37,252 # Manual UART driver test application
2025-01-07 07:04:37,253 # ===================================
2025-01-07 07:04:37,258 # This application is intended for testing additional UART
2025-01-07 07:04:37,264 # interfaces, that might be defined for a board. The 'primary' UART
2025-01-07 07:04:37,270 # interface is tested implicitly, as it is running the shell...
2025-01-07 07:04:37,271 # 
2025-01-07 07:04:37,276 # When receiving data on one of the additional UART interfaces, this
2025-01-07 07:04:37,282 # data will be outputted via STDIO. So the easiest way to test an 
2025-01-07 07:04:37,288 # UART interface, is to simply connect the RX with the TX pin. Then 
2025-01-07 07:04:37,294 # you can send data on that interface and you should see the data 
2025-01-07 07:04:37,295 # being printed to STDOUT
2025-01-07 07:04:37,295 # 
2025-01-07 07:04:37,301 # NOTE: all strings need to be '\n' terminated!
2025-01-07 07:04:37,301 # 
2025-01-07 07:04:37,558 # UARD_DEV(0): test uart_poweron() and uart_poweroff()  ->  [OK]
2025-01-07 07:04:37,559 # 
2025-01-07 07:04:37,559 # UART INFO:
2025-01-07 07:04:37,561 # Available devices:               4
2025-01-07 07:04:37,566 # UART used for STDIO (the shell): UART_DEV(0)
2025-01-07 07:04:37,566 # 

2025-01-07 07:04:38,945 # test 1
2025-01-07 07:04:38,946 # [START]
2025-01-07 07:04:39,077 # [SUCCESS]

2025-01-07 07:05:04,084 # test 2
2025-01-07 07:05:04,085 # [START]
2025-01-07 07:05:04,216 # [SUCCESS]

2025-01-07 07:05:31,107 # test 3
2025-01-07 07:05:31,108 # [START]
2025-01-07 07:05:31,240 # [SUCCESS]
```
</details>

<details><summary>tests/periph/spi</summary>

```
2025-01-07 07:09:39,003 # main(): This is RIOT! (Version: 2025.01-devel-342-gfad857-cpu/same51)
2025-01-07 07:09:39,010 # Manual SPI peripheral driver test (see README.md)
2025-01-07 07:09:39,015 # There are 1 SPI devices configured for your platform.

2025-01-07 07:09:42,588 # init 0 0 0
2025-01-07 07:09:42,591 # Trying to initialize SPI_DEV(0): mode: 0, clk: 0, cs_port: 0, cs_pin: 0
2025-01-07 07:09:42,603 # (if below the program crashes with a failed assertion, then it means the configuration is not supported)
2025-01-07 07:09:42,603 # Success.

2025-01-07 07:09:44,361 # send test
2025-01-07 07:09:44,363 # Sent bytes
2025-01-07 07:09:44,364 #    0    1    2    3 
2025-01-07 07:09:44,366 #   0x74 0x65 0x73 0x74
2025-01-07 07:09:44,367 #     t    e    s    t 
2025-01-07 07:09:44,368 # 
2025-01-07 07:09:44,369 # Received bytes
2025-01-07 07:09:44,370 #    0    1    2    3 
2025-01-07 07:09:44,379 #   0x74 0x65 0x73 0x74
2025-01-07 07:09:44,380 #     t    e    s    t 
2025-01-07 07:09:44,380 # 
```
</details>

<details><summary>tests/periph/i2c</summary>

```
```
</details>

### Issues/PRs references
